File: PR1/PR1_Ej3/HamburguesaVeganaBuilder.py
from HamburguesaBuilder import HamburguesaBuilder
import time
class HamburguesaVeganaBuilder(HamburguesaBuilder):

    def aniadePan(self):
        self.hamburguesa.pan = "Pan normal"
        time.sleep(0.1)

    def aniadeLechuga(self):
        self.hamburguesa.lechuga = "Lechuga fresca"
        time.sleep(0.1)

    def aniadeTomate(self):
        self.hamburguesa.tomate = "Tomate pera"
        time.sleep(0.1)

    def aniadeCebolla(self):
        self.hamburguesa.cebolla = "Cebolla llorosa"
        time.sleep(0.1)

    def aniadePepinillos(self):
        self.hamburguesa.pepinillos = "Pepinillos"
        time.sleep(0.1)

    def aniadeBacon(self):
        self.hamburguesa.bacon = "Bacon Vegano reseco"
        time.sleep(0.6)

    def aniadeCarne(self):
        self.hamburguesa.carne = "Carne Vegana de dudosa procedencia"
        time.sleep(1)

    def aniadePrecio(self):
        self.hamburguesa.precio = 5.5

File: PR2/pantalla_pedidos_hamburgueseria/lib/HamburguesaVeganaBuilder.dart
import 'HamburguesaBuilder.dart';
import 'Hamburguesa.dart';


class HamburguesaVeganaBuilder extends HamburguesaBuilder {
  @override
  Hamburguesa hamburguesa = Hamburguesa() ;

  @override
  void aniadePan() {
    hamburguesa.pan = "Pan normal";
    sleep();
  }

  @override
  void aniadeLechuga() {
    hamburguesa.lechuga = "Lechuga fresca";
    sleep();
  }

  @override
  void aniadeTomate() {
    hamburguesa.tomate = "Tomate pera";
    sleep();
  }

  @override
  void aniadeCebolla() {
    hamburguesa.cebolla = "Cebolla llorosa";
    sleep();
  }

  @override
  void aniadePepinillos() {
    hamburguesa.pepinillos = "Pepinillos";
    sleep();
  }

  @override
  void aniadeBacon() {
    hamburguesa.bacon = "Bacon Vegano reseco";
    sleepLong();
  }

  @override
  void aniadeCarne() {
    hamburguesa.carne = "Carne Vegana de dudosa procedencia";
    sleepLong();
  }

  @override
  void aniadePrecio() {
    hamburguesa.precio = 5.5;
  }
}